com.unity.multiplayer.mlapi/Tests/Runtime/NetworkPrefabHandlerTests.cs

```diff
@@ -0,0 +1,162 @@
+
+            //Register via NetworkObject
+            gameObjectRegistered = networkPrefabHandler.AddHandler(baseObject, networkPrefaInstanceHandler);
+
+            //Test result of registering via NetworkObject reference
+            Assert.True(gameObjectRegistered);
+
+            //Change it up
+            prefabPosition = new Vector3(2.0f, 1.0f, 5.0f);
+            prefabRotation = new Quaternion(4.0f, 1.5f, 5.4f, 5.1f);
+
+            spawnedObject = networkPrefabHandler.HandleNetworkPrefabSpawn(baseObject.GlobalObjectIdHash, 0, prefabPosition, prefabRotation);
+
+            //Test that something was instantiated
+            Assert.NotNull(spawnedObject);
+
+            //Test that this is indeed an instance of our original object
+            Assert.True(spawnedObject.name.Contains("NetworkPrefabHandlerTestObject"));
+
+            //Test for position and rotation
+            Assert.True(prefabPosition == spawnedObject.transform.position);
+            Assert.True(prefabRotation == spawnedObject.transform.rotation);
+
+            networkPrefabHandler.HandleNetworkPrefabDestroy(spawnedObject);     //Destroy our prefab instance
+            networkPrefabHandler.RemoveHandler(baseObject);                     //Remove our handler
+
+            //Register via GlobalObjectHashId
+            gameObjectRegistered = networkPrefabHandler.AddHandler(baseObject.GlobalObjectIdHash, networkPrefaInstanceHandler);
+
+            //Test result of registering via GlobalObjectHashId reference
+            Assert.True(gameObjectRegistered);
+
+            //Change it up
+            prefabPosition = new Vector3(6.0f, 4.0f,1.0f);
+            prefabRotation = new Quaternion(3f, 2f, 4f, 1f);
+
+            spawnedObject = networkPrefabHandler.HandleNetworkPrefabSpawn(baseObject.GlobalObjectIdHash, 0, prefabPosition, prefabRotation);
+
+            //Test that something was instantiated
+            Assert.NotNull(spawnedObject);
+
+            //Test that this is indeed an instance of our original object
+            Assert.True(spawnedObject.name.Contains("NetworkPrefabHandlerTestObject"));
+
+            //Test for position and rotation
+            Assert.True(prefabPosition == spawnedObject.transform.position);
+            Assert.True(prefabRotation == spawnedObject.transform.rotation);
+
+            networkPrefabHandler.HandleNetworkPrefabDestroy(spawnedObject);     //Destroy our prefab instance
+            networkPrefabHandler.RemoveHandler(baseObject);                     //Remove our handler
+
+            Assert.False(networkPrefaInstanceHandler.StillHasInstances());
+        }
+
+        public void Dispose()
+        {
+            //Stop, shutdown, and destroy
+            NetworkManagerHelper.ShutdownNetworkManager();
+        }
+
+        public NetworkPrefabHandlerTests()
+        {
+            //Create, instantiate, and host
+            NetworkManagerHelper.StartNetworkManager();
+        }
```

Comment:
I don't know where our previous discussion went but I believe we should stick with `[SetUp]` and `[TearDown]` attributed methods for construction and destruction of objects / components within tests.

I barely remember you mentioning `NetworkManager` instance being used by more than one test during Unity test session and I'd even go as far as saying that it's a bad practice which we should drop.

Tests should be as much isolated as they can. Objects, components or any other artifacts left from another test run might effect another test, that's something we should absolutely avoid. Re-using the same NetworkManager instance / gameobject breaks that isolation and sandbox environment for tests to be isolated and us to be confident about our automated tests.

I'm happy to discuss further internally or here if you'd prefer.

Reply:
Working on coming up with a solution for this.

Reply:
It is now applied.